llvm/test/CodeGen/AMDGPU/vgpr-lowering-gfx1250.mir

```diff
@@ -480,7 +480,7 @@ body:             |
     ; GCN-NEXT: v_mov_b32_e32 v0 /*v256*/, v1
     $vgpr256 = V_MOV_B32_e32 undef $vgpr1, implicit $exec
 
-    ; No mode switch on fall through
+    ; No mode switch on fall through if this is the only predecessor
```

Comment:
Technically this small optimization also breaks the convention, although in a real compilation blocks will be squashed together. Maybe optimization is not needed here at all.